### ISSUE TITLE
Stop using scope.lookupvar() in templates

### ISSUE THREAD

**Issue by ghoneycutt** [sensu/sensu-puppet#701]
We should bring the variables into the local scope and use `@variable`

https://docs.puppet.com/puppet/4.10/lang_template_erb.html#accessing-puppet-variables

**Comment by Phil-Friderici** [comment]
If we move the variables to the classes where they are really used, we will break backward compatibility.
For example the template sensu.erb is used in sensu::package but the variables are sourced in from the main class.

To maintain backward compatibility in the first step, I could add the variables to sensu::package and source them in from the main class. This way you get the same results using the old way.

``` puppet
class sensu::package (
  $log_dir = $::sensu::log_dir,
  ) {
```

**Comment by ghoneycutt** [comment]
Let's go the second route and do variable assignment that references variables in the main class. Eventually I'd like to collapse all these extra classes and this will make it easier to understand where the variables come from.

**Comment by Phil-Friderici** [comment]
Working on this in PR https://github.com/sensu/sensu-puppet/pull/724

**Comment by Phil-Friderici** [comment]
this issue can be closed now :)

**Comment by ghoneycutt** [comment]
Released in v2.22.0
